packages/axelar-soroban-std/src/events.rs

```diff
@@ -0,0 +1,155 @@
+        ) -> bool;
+
+        fn standardized_fmt(
+            env: &Env,
+            event: &(soroban_sdk::Address, soroban_sdk::Vec<Val>, Val),
+        ) -> std::string::String;
+    }
+
+    pub fn fmt_last_emitted_event<E>(env: &Env) -> std::string::String
+    where
+        E: EventTestutils,
+    {
+        let event = env.events().all().last().expect("no event found");
+        E::standardized_fmt(env, &event)
+    }
+
+    pub fn fmt_emitted_event_at_idx<E>(env: &Env, idx: u32) -> std::string::String
+    where
+        E: EventTestutils,
+    {
+        let event = env
+            .events()
+            .all()
+            .get(idx)
+            .expect("no event found at the given index");
+        E::standardized_fmt(env, &event)
+    }
+
+    #[macro_export]
+    macro_rules! impl_event_testutils {
+        ($event_type:ty, ($($topic_type:ty),*), ($($data_type:ty),*)) => {
+            impl $crate::events::EventTestutils for $event_type {
+                fn matches(&self, env: &soroban_sdk::Env, event: &(soroban_sdk::Address, soroban_sdk::Vec<soroban_sdk::Val>, soroban_sdk::Val)) -> bool {
+                    use soroban_sdk::IntoVal;
+
+                    Self::standardized_fmt(env, event) == Self::standardized_fmt(env, &(event.0.clone(), self.topics().into_val(env), self.data().into_val(env)))
+                }
+
+                #[allow(unused_assignments)]
+                fn standardized_fmt(env: &soroban_sdk::Env, (contract_id, topics, data): &(soroban_sdk::Address, soroban_sdk::Vec<soroban_sdk::Val>, soroban_sdk::Val)) -> std::string::String {
+                    use soroban_sdk::TryFromVal;
+
+                    let mut topics_output = std::vec![];
+
+                    let mut i = 0;
+                    $(
+                        let topic = topics.get(i).expect("the number of topics does not match this function's definition");
+                        topics_output.push(std::format!("{:?}", <$topic_type>::try_from_val(env, &topic)
+                            .expect("given topic value does not match the expected type")));
+
+                        i += 1;
+                    )*
+
+                    let data = soroban_sdk::Vec::<soroban_sdk::Val>::try_from_val(env, data).expect("data should be defined as a vector-compatible type");
+
+                    let mut data_output = std::vec![];
+
+
+                    let mut i = 0;
+                    $(
+                        let data_entry = data.get(i).expect("the number of data entries does not match this function's definition");
+                        data_output.push(std::format!("{:?}", <$data_type>::try_from_val(env, &data_entry)
+                            .expect("given data value does not match the expected type")));
+
+                        i += 1;
+                    )*
+
+                    std::format!("contract: {:?}, topics: ({}), data: ({})", contract_id, topics_output.join(", "), data_output.join(", "))
```

Comment:
print on different lines
```suggestion
                    std::format!("contract: {:?}\n\ntopics: ({})\n\ndata: ({})", contract_id, topics_output.join(", "), data_output.join(", "))
```